ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -580,3 +581,28 @@ def test_ipa_certs_check_ipacertnsstrust(self):
                 check["kw"]["key"] in DEFAULT_PKI_KRA_CERTS
             )
         tasks.uninstall_master(self.master)
+
+
+class TestIpaHealthCheckOnReplica(IntegrationTest):
+
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_dns=True)
+        tasks.install_replica(cls.master, cls.replicas[0], setup_dns=True)
+
+    def test_run_with_stopped_master(self):
+        """
+        Test output of healthcheck where master IPA services are stopped
+        contains only errors regarding master being stopped and no other false
+        positives.
+        """
+        self.master.run_command(["ipactl", "stop"])
+        returncode, output = run_healthcheck(
+            self.master,
+            output_type="human",
+            failures_only=True
+        )
+        errors = re.findall("^ERROR: .*: not running$", output)
+        assert len(errors) == len(output.split('\n'))
```

Comment:
Working on a proposal. I think the current patch just needs some tweaking to pass. What I'm doing is:

- dropping as a separate class and integrating into TestIpaHealthcheck
- adding a fixture to restart IPA once the test is complete
- dropping the ^ and $ from the output and replacing \n with '' so the re will succeed